lib/ansible/modules/cloud/vmware/vmware_local_user_manager.py

```diff
@@ -111,71 +144,243 @@ def process_state(self):
+                "on the root node : %s" % to_native(admin_permission)
+            )
+        except vmodl.fault.ManagedObjectNotFound as object_not_found:
+            self.module.fail_json(msg="The specified entity doesn't exist : %s" % to_native(object_not_found))
+        except vmodl.fault.InvalidArgument as invalid_argument:
+            self.module.fail_json(
+                msg="The new role ID is the View or Anonymous role, "
+                "or the entity does not support assigning permissions : %s" % to_native(invalid_argument)
+            )
+        except vim.fault.NoPermission as no_permission:
+            self.module.fail_json(msg="No priviledge to set permission : %s" % to_native(no_permission))
+
+    @staticmethod
+    def get_role_id(role_name, role_list):
+        """Get role ID from role name"""
+        role_id = None
+        # Default roles
+        if role_name == 'NoAccess':
+            role_id = -5
+        elif role_name == 'ReadOnly':
+            role_id = -2
+        elif role_name == 'Admin':
+            role_id = -1
+        # Custom roles
+        else:
+            for role in role_list:
+                if role.name == role_name:
+                    role_id = role.roleId
+                    break
+        return role_id
+
+    @staticmethod
+    def get_role_name(role_id, role_list):
+        """Get role name from role ID"""
+        role_name = None
+        # Default roles
+        if role_id == -5:
+            role_name = 'NoAccess'
+        elif role_id == -2:
+            role_name = 'ReadOnly'
+        elif role_id == -1:
+            role_name = 'Admin'
+        # Custom roles
+        else:
+            for role in role_list:
+                if role.roleId == role_id:
+                    role_name = role.name
+                    break
+        return role_name
+
+    def state_create_user(self):
+        """Create the user account"""
+        results = dict()
+        results['user_name'] = self.local_user_name
+        results['description'] = self.local_user_description
+        results['role'] = self.local_user_role
+        if self.module.check_mode:
+            results['msg'] = "User account would be created"
+        else:
+            account_spec = self.create_account_spec()
+            try:
+                self.content.accountManager.CreateUser(account_spec)
+            except vim.fault.AlreadyExists as already_exists:
+                self.module.fail_json(
+                    msg="The specified local user account already exists : %s" % to_native(already_exists)
+                )
+            except vmodl.fault.InvalidArgument as invalid_argument:
+                self.module.fail_json(
+                    msg="The user name or password has an invalid format : %s" % to_native(invalid_argument)
+                )
+            if self.local_user_role:
+                role_id = self.get_role_id(self.local_user_role, self.content.authorizationManager.roleList)
```

Comment:
Add condition here to check if role_id is `None` or not. If I specify `admin` as `local_user_role` then I get 

```
fatal: [localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "hostname": "10.76.33.226",
            "local_user_description": null,
            "local_user_name": "aa",
            "local_user_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "local_user_role": "admin",
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "port": 443,
            "propagate": true,
            "state": "present",
            "update_password": false,
            "username": "root",
            "validate_certs": false
        }
    },
    "msg": "Required field \"roleId\" not provided (not @optional)"
}
```

Reply:
done. thx.